lib/3.7/packages.cf

```diff
@@ -1229,8 +1229,8 @@ body package_method smartos
 {
       package_changes => "bulk";
       package_list_command => "/opt/local/bin/pkgin list";
-      package_list_name_regex    => "([^\s]+)\-[0-9]+.*\s";
-      package_list_version_regex => "[^\s]+\-([0-9][^\s]+)\s";
+      package_list_name_regex    => "([^\s]+)\-[0-9a-b.]+.*[; ]";
+      package_list_version_regex => "[^\s]+\-([0-9a-b.]+).*[; ]";
```

Comment:
Is it perhaps more future proof to keep using the ``[^\s]``, but include the ``;`` in the rejected set, like:

<pre>
package_list_name_regex    => "([^\s]+)\-[0-9][^\s;]+.*[\s;]";
package_list_version_regex => "[^\s]+\-([0-9][^\s;]+).*[\s;]";
</pre>

This has a couple of advantages:

1. You guarantee that version starts with a number, which the old regex did ensure.
1. I can see versions potentially containing dashes or underscores, which would fail to match your regex, but should match the one above.
1. It will handle tabs correctly if the package manager ever decides to output one of those.